edpop_explorer/readers/stcn.py

```diff
@@ -107,6 +138,69 @@ def _convert_record(cls, rawrecord: dict) -> BiographicalRecord:
         return record
 
 
+class STCNPrintersReader(STCNBaseReader):
+    API_BY_ID_BASE_URL = 'https://data.cerl.org/stcn_printers/'
+    LINK_BASE_URL = 'https://data.cerl.org/stcn_printers/'
+    CATALOG_URIREF = URIRef(
+        'https://edpop.hum.uu.nl/readers/stcn-printers'
+    )
+    IRI_PREFIX = "https://edpop.hum.uu.nl/readers/stcn-printers/"
+    READERTYPE = BIOGRAPHICAL
+    SHORT_NAME = "STCN Printers"
+    DESCRIPTION = "National bibliography of The Netherlands until 1801 – printers"
+
+    @classmethod
+    def transform_query(cls, query) -> str:
+        # Only person records
+        return f"({query}) AND data.type:impr"
+
+    @classmethod
+    def _get_name(cls, rawrecord: dict) -> Optional[Field]:
+        display_name = safeget(rawrecord, ('shortDisplay',))
+        if display_name:
+            return Field(' '.join(reversed(display_name.split(', '))))
+
+    @classmethod
+    def _get_places_of_activity(cls, rawrecord: dict) -> Optional[List[Field]]:
+        places = safeget(rawrecord, ('data', 'place',))
+        if places:
+            return [Field(x['text']) for x in places]
```

Comment:
This pattern is repeated several times below and also on line 71 of `sbtireader.py` and possibly other places. I suggest outfactoring it.